content/docs/user-guide/dvc-files.md

```diff
@@ -237,14 +242,224 @@ stages:
 ```
 
 Stage commands are listed again in `dvc.lock`, in order to know when their
-definitions change in the `dvc.yaml` file.
+definitions change in `dvc.yaml`.
 
 Regular <abbr>dependencies</abbr> and all kinds of <abbr>outputs</abbr>
 (including [metrics](/doc/command-reference/metrics) and
 [plots](/doc/command-reference/plots) files) are also listed (per stage) in
-`dvc.lock`, but with an additional field to store the hash value of each file or
-directory tracked by DVC. Specifically: `md5`, `etag`, or `checksum` (same as in
-`deps` and `outs` entries of `.dvc` files).
+`dvc.lock`, but with an additional field with a hash of their last known
+contents. Specifically: `md5`, `etag`, or `checksum` are used (same as in `deps`
+and `outs` entries of `.dvc` files).
+
+Full <abbr>parameters</abbr> (key and value) are listed too (under `params`),
+grouped by parameters file. And in the case of templated/parameterized
+`dvc.yaml` files (see next section), their actual values are substituted into
+the `dvc.lock` YAML structure.
+
+### Parameterize `dvc.yaml`
+
+`dvc.yaml` supports a special templating format that enables reusing simple
+values from different sources in the YAML structure itself.
+
+Global parameters (file scope) can be defined either in a
```

Comment:
what is `file scope`? what is `global`?

Better start with "Let's say .." - give an example, explain things after

Reply:
Done 🙂